common/network-common/src/main/java/org/apache/spark/network/server/TransportChannelHandler.java

```diff
@@ -158,10 +158,12 @@ public void userEventTriggered(ChannelHandlerContext ctx, Object evt) throws Exc
       // To avoid a race between TransportClientFactory.createClient() and this code which could
       // result in an inactive client being returned, this needs to run in a synchronized block.
       synchronized (this) {
+        // Do not modify the order of hasInFlightRequests and isActuallyOverdue (see SPARK-27073)
+        boolean hasInFlightRequests = responseHandler.hasOutstandingRequests();
         boolean isActuallyOverdue =
           System.nanoTime() - responseHandler.getTimeOfLastRequestNs() > requestTimeoutNs;
         if (e.state() == IdleState.ALL_IDLE && isActuallyOverdue) {
-          if (responseHandler.hasOutstandingRequests()) {
+          if (hasInFlightRequests) {
```

Comment:
Could you explain why "Do not modify the order of hasInFlightRequests and isActuallyOverdue " please ?

Reply:
Are you referring to adding more comments to the code, or explaining why this happens in this PR?

Original PR #23989 should explain:
> When TransportChannelHandler processes IdleStateEvent, it first calculates whether the last request time has timed out.
At this time, TransportClient.sendRpc initiates a request.
TransportChannelHandler gets responseHandler.numOutstandingRequests() > 0, causing the normal connection to be closed.